src/keys/msgsig.rs

```diff
@@ -0,0 +1,128 @@
+
+/// Signing arbitrary messages and verify message signatures
+
+use bitcoin::hashes::Hash;
+use bitcoin::secp256k1::recovery::RecoverableSignature;
+use bitcoin::secp256k1::{All, Message, Secp256k1, SecretKey};
+use bitcoin::util::misc::{signed_msg_hash, MessageSignature};
+use bitcoin::{Address, Network, PrivateKey, PublicKey};
+
+/// Trait for message signers
+pub trait MessageSigner<S> {
+    /// Sign an arbitrary message
+    fn sign(&self, msg: &str) -> S;
+}
+
+/// Trait for message signature verifiers
+pub trait MessageSignatureVerifier<S> {
+    /// Verify a signature against an arbitrary message
+    fn verify(&self, sig: S, msg: &str) -> bool;
+}
+
+/// A message signer using ECDSA
+pub struct EcdsaMessageSigner {
+    secp: Secp256k1<All>,
```

Comment:
Same thing about secp as for the verifier

Reply:
`secp` as a reference: https://github.com/bitcoindevkit/bdk/pull/601/commits/707f6c3de09a8e1d3cf2b8cd6401724f7490104f